### ISSUE TITLE
[BUG] time_to_idle not working as expected

### ISSUE THREAD

**Issue by sattva9** [moka-rs/moka#359]
`time_to_idle` function giving unexpected results. Follow below steps to replicate the issue.

```
[dependencies]
moka = { version = "0.12.1", features = ["sync"] }
```

```
fn main() {
    let cache = moka::sync::Cache::builder()
        .max_capacity(10)
        .time_to_idle(std::time::Duration::from_secs(5))
        .build();

    cache.insert(1, 1);

    std::thread::sleep(std::time::Duration::from_secs(4));
    println!("{:?}", cache.get(&1));

    std::thread::sleep(std::time::Duration::from_secs(2));
    println!("{:?}", cache.get(&1));
    println!("{:?}", cache.get(&1));
}

```

**Output**
```
Some(1)
None
Some(1)

```
**Expected Output**
```
Some(1)
Some(1)
Some(1)
```

**Comment by peter-scholtens** [comment]
Confirmed, and also present in branch _fix-race-in-handle-upseart_. Not present in __v0.10.4__ and __v.0.11.3__.

**Comment by tatsuya6502** [comment]
Thanks for reporting the issue and confirming in different versions!

I believe this issue should be present in v0.11.x and older versions too, but it should be harder to reproduce in those versions as they had background threads to periodically process pending tasks. Anyway, I think I can fix it for v0.12.x.

**Comment by tatsuya6502** [comment]
FYI, I published moka [v0.12.2](https://crates.io/crates/moka/0.12.2) with the fix to crates.io.
